include/highfive/h5easy_bits/H5Easy_misc.hpp

```diff
@@ -59,16 +48,144 @@ inline void createGroupsToDataSet(File& file, const std::string& path) {
     }
 }
 
-}  // namespace detail
+// Generate error-stream and return "Exception" (not yet thrown).
+inline Exception error(const File& file,
+                       const std::string& path,
+                       const std::string& message) {
+    std::ostringstream ss;
+    ss << message << std::endl
+       << "Path: " << path << std::endl
+       << "Filename: " << file.getName() << std::endl;
+    return Exception(ss.str());
+}
+
+// Generate specific dump error
+inline Exception dump_error(File& file, const std::string& path)
+{
+    if (file.getObjectType(path) == ObjectType::Dataset) {
+        return error(file, path,
+            "H5Easy: Dataset already exists, dump with H5Easy::DumpMode::Overwrite "
+            "to overwrite (with an array of the same shape).");
+    } else {
+        return error(file, path,
+            "H5Easy: path exists, but does not correspond to a Dataset. Dump not possible.");
+    }
+}
 
-inline size_t getSize(const File& file, const std::string& path) {
-    return file.getDataSet(path).getElementCount();
+// get a opened DataSet: nd-array
+template <class T>
+inline DataSet init_dataset(File& file,
+                            const std::string& path,
+                            const std::vector<size_t>& shape,
+                            const DumpOptions& options)
+{
+    if (!file.exist(path)) {
+        detail::createGroupsToDataSet(file, path);
+        if (!options.isCompress()) {
```

Comment:
Naive question: can I have a chuncked dataset without compression?

Reply:
I suppose (but honestly this lies outside my normal use case). Anyways I could easily allow it